Automattic/wp-calypso
# Launchpad: Move share site modal to launchpad package

Depends on: https://github.com/Automattic/wp-calypso/pull/82883

## Proposed Changes

* This PR moves the Share Site Modal component to the Launchpad package. We'll need it in the package so we can access it on the jetpack-mu-plugin.

We should deploy https://github.com/Automattic/wp-calypso/pull/82883 before this PR.

## Testing Instructions



* We need to ensure the modal continues to work as expected on the Customer Home launchpad
* Use the Calypso Live link below
* On a site with the Build intent, navigate to the Customer Home and click on the "Share your site" task
* The modal should appear and you should be able to copy the site link

## Pre-merge Checklist
